### PR TITLE
Add wayfinding components

### LINKED ISSUE #2: How to know your location and navigate the site
_As a user of the design system
I want to know how to communicate location and navigation
So that there is a consistent experience across sites_

## Description

The aim of a global navigation is to:
- switch between top-level categories easily, no matter their current location
- get a sense of what is available on the website even if they don't enter via the homepage
- communicate the brand

## Links and resources

- Related global footer and header spike: https://github.com/barnardos/digital-transformation/issues/80

## Acceptance Criteria

- [x] Addresses accessibility/usability considerations
- [x] Works on target devices and browsers

Elements to consider:
- [x] Rules for Logo (homepage versus drilled down pages)
- [x] Global navigation/menu/when to display navigation v when to collapse
- [ ] Rules for primary/secondary/tertiary items
- [x] Adaptation for smaller viewports, while retaining the underlying universal rules
- [x] Search
- [ ] Sign in/Sign up admin links
- [ ] With/without imagery
- [x] Adaptations for microsite
- [x] Responsive considerations

### PR DESCRIPTION
**Pull request by jeddy3**
Closes #2

### CODE REVIEW

File: src/components/SubmitSearchControl/index.css [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+.SubmitSearchControl-commands {
+  display: flex;
+}
+
+.SubmitSearchControl-command {
+  appearance: none;
+  background: #e9e9e9 url("./glass.svg") no-repeat center center;
+  border-radius: 0;
+  box-shadow: 0.125rem 0.125rem #a5a5a5;
+  color: #ffffff;
```
Comment (tosbourn) [inline src/components/SubmitSearchControl/index.css]: In other places we shorted 6 identical values to 3.

``` diff
- color: #ffffff;
+ color: #fff;
```

File: src/pages/index.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import Helmet from "react-helmet";
 import React from "react";
 
-import Navigation from "../components/Navigation";
+import DesignSystemNavigation from "../components/DesignSystemNavigation";
```
Comment (tosbourn) [inline src/pages/index.js]: Can you explain why we're calling this the `DesignSystemNavigation` but below isn't `DesignSystemTitle`?

Comment (jeddy3) [reply src/pages/index.js]: Good question. This navigation is specific to the Design System _website_. It's the list of sections on the landing pages. Whereas `Title` is both a part of the Design System itself and used on the Design System _website_

File: src/components/SubmitSearchControl/index.css (hunk 1 above)
Comment (jeddy3) [reply src/components/SubmitSearchControl/index.css]: It looks like it's inconsistent throughout. I'll create a separate issue for it.

File: src/components/Navigation/index.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      isRevealing ? "Navigation--revealing" : ""
+    }`;
+    const detailsClassName = `Navigation-details ${
+      isActive ? "Navigation-details--active" : ""
+    }`;
+    return (
+      <nav className={navigationClassName}>
+        {isRevealing && (
+          <button className="Navigation-command" onClick={this.handleClick}>
+            Menu
```
Comment (joelcarr) [inline src/components/Navigation/index.js]: From my understanding, this text is hidden and the hamburger icon replaces it via CSS. Is this accessible? Is the hamburger on its own understood enough to be a menu?

File: src/components/Navigation/index.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    const navigationClassName = `Navigation ${
+      isRevealing ? "Navigation--revealing" : ""
+    }`;
+    const detailsClassName = `Navigation-details ${
+      isActive ? "Navigation-details--active" : ""
+    }`;
+    return (
+      <nav className={navigationClassName}>
+        {isRevealing && (
+          <button className="Navigation-command" onClick={this.handleClick}>
```
Comment (joelcarr) [inline src/components/Navigation/index.js]: I have recently found out about the [`details`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details) tag which GitHub uses for their 'click to reveal' menus. Should we incorporate this practice?

File: src/components/Breadcrumbs/index.css [hunk 5]
```diff
@@ -0,0 +1,16 @@
+.Breadcrumbs-items {
+  display: flex;
```
Comment (joelcarr) [inline src/components/Breadcrumbs/index.css]: We need to ensure the layout breaks to multiple rows when on smaller viewports:

image

File: src/components/Navigation/index.js (hunk 4 above)
Comment (jeddy3) [reply src/components/Navigation/index.js]: > Should we incorporate this practice?

There are limitations e.g. it's not possible to alter the animation. So, I went with a custom implementation (but used the "details" name).

File: src/components/Navigation/index.js (hunk 3 above)
Comment (jeddy3) [reply src/components/Navigation/index.js]: > this text is hidden and the hamburger icon replaces it via CSS. Is this accessible?

It's hidden using `text-ident` so VoiceOver reads it.

> Is the hamburger on its own understood enough to be a menu?

That's unclear. There's a follow up issue to test this (amongst other aspects of the DS) with service users: https://github.com/barnardos/design-system/issues/95

File: src/components/Breadcrumbs/index.css (hunk 5 above)
Comment (jeddy3) [reply src/components/Breadcrumbs/index.css]: 👍

**Pull request merged**